tests/unit/s2n_signature_algorithms_test.c

```diff
@@ -345,6 +345,27 @@ int main(int argc, char **argv)
                     S2N_ERR_INVALID_SIGNATURE_SCHEME);
         }
 
+        /* Test: no shared valid signature schemes, using TLS1.2 */
+        {
+            conn->secure.cipher_suite = TLS13_CIPHER_SUITE;
+            conn->actual_protocol_version = S2N_TLS12;
+
+            struct s2n_sig_scheme_list peer_list = {
+                    .len = 1, .iana_list = {
+                            s2n_ecdsa_secp384r1_sha384.iana_value, /* Invalid (wrong protocol version) */
+                    },
```

Comment:
It looks like this is testing the case when the signature scheme list offered doesn't match the protocol version negotiated. Would it also be worth testing just the 1.2 on 1.2 case? E.g.
- s2n_cipher_suite s2n_dhe_rsa_with_chacha20_poly1305_sha256
- conn->actual_protocol_version = S2N_TLS12
- s2n_signature_scheme s2n_ecdsa_sha512

Reply:
Cool, that sounds fine to me. 

Another question just to make sure I'm understanding the test: We call `s2n_choose_sig_scheme_from_peer_preference_list` when we process the client hello but we don't explicitly call `s2n_choose_default_sig_scheme` like this test does. https://github.com/awslabs/s2n/blob/b7611731a97fda1fc765660bb1a53a6747984781/tls/s2n_client_hello.c#L312-L321

Does `s2n_choose_default_sig_scheme` get called in one of those other functions? If so, I don't think this test guarantees that will always be true. We're still vulnerable to the case where someone changes one of these functions and the overall behavior of the handshake changes.

Reply:
s2n_choose_default_sig_scheme() is called as part of s2n_choose_sig_scheme_from_peer_preference_list(), the function that we're testing.

We only call s2n_choose_default_sig_scheme() in the test to set the value of "default_scheme", which we then compare to the scheme set by calling s2n_choose_sig_scheme_from_peer_preference_list().

But it sounds like this test could use more comments.

Reply:
Per offline discussion, we'll figure out a way to test the end-to-end experience in a separate PR: https://github.com/awslabs/s2n/issues/1632